src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -135,7 +135,7 @@ public bool ChildContentRetrieved
         /// <summary>
         /// <see cref="ITagHelper"/>s that should be run.
         /// </summary>
-        public IEnumerable<ITagHelper> TagHelpers
+        public IList<ITagHelper> TagHelpers
```

Comment:
Ah, so I thought about that. The issue with that approach is that `TagHelper` authors wouldn't be able to determine their `Order` dynamically based on their provided properties with how we code gen today.

Ex:
```C#
 __TestNamespace_InputTagHelper = CreateTagHelper<global::TestNamespace.InputTagHelper>();
__tagHelperExecutionContext.Add(__TestNamespace_InputTagHelper); // Would be ordered on Insert, in this case we don't have any properties set yet.
__TestNamespace_InputTagHelper2 = CreateTagHelper<global::TestNamespace.InputTagHelper2>();
__tagHelperExecutionContext.Add(__TestNamespace_InputTagHelper2);
__TestNamespace_InputTagHelper.Type = "checkbox";
```

It'd be a larger change to transition the `.Add` after the attribute initialization of the `TagHelper`s. Wanted to avoid the larger change. If you feel like it has a lot of value I can do it.